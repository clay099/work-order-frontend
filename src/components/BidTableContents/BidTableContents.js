import React from "react";
import { TableBody, TableRow, TableCell, Button } from "@material-ui/core/";
import { Send } from "@material-ui/icons/";
import { acceptBidWithAPI } from "../../actions/bids";
import { useDispatch, useSelector } from "react-redux";
import { useHistory } from "react-router-dom";

const BidTableContents = ({ bidData, projectId }) => {
	const { token } = useSelector((st) => ({ token: st.login.token }));
	const dispatch = useDispatch();
	const history = useHistory();

	const handleSubmit = async ({ tradesmenId, price }) => {
		let resp = await dispatch(
			acceptBidWithAPI({
				projectId,
				token,
				tradesmen_id: +tradesmenId,
				price,
			})
		);
		history.push("/user");
	};

	return (
		<TableBody>
			{Object.keys(bidData).map((tradesmenId) => {
				console.log(bidData[tradesmenId]);
				return (
					<TableRow key={bidData[tradesmenId]}>
						<TableCell>{bidData[tradesmenId].first_name}</TableCell>
						<TableCell>{bidData[tradesmenId].last_name}</TableCell>
						<TableCell>
							{Number(bidData[tradesmenId].bid).toLocaleString(
								"en-US",
								{
									style: "currency",
									currency: "USD",
								}
							)}
						</TableCell>
						<TableCell>
							<Button
								type="submit"
								size="small"
								variant="outlined"
								color="primary"
								endIcon={<Send />}
								onClick={() =>
									handleSubmit({
										tradesmenId,
										price: bidData[tradesmenId].bid,
									})
								}
							>
								Accept Bid
							</Button>
						</TableCell>
					</TableRow>
				);
			})}
		</TableBody>
	);
};

export default BidTableContents;
