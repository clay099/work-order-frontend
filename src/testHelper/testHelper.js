// testHelper.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
// Import your own reducer
import reducer from "../reducers/rootReducer";

function render(
	ui,
	{
		initialState,
		store = createStore(reducer, initialState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const tokenState = { token: "token" };
const userTypeUserState = { userType: "user" };
const userTypeTradesmenState = { userType: "user" };
const idState = { id: 1234 };
const userDetailsState = { userDetails: "userDetails" };
const handleSubmitState = { handleSubmit: jest.fn() };
const handleChangeState = { handleChange: jest.fn() };
const disabledState = { disabled: false };
const disabledEmailState = { disabledEmail: false };
const buttonTextState = { buttonText: "buttonText" };
const formDataState = {
	formData: {
		id: 1,
		firstName: "firstName",
		lastName: "lastName",
		email: "email",
		password: "password",
		newPassword: "newPassword",
		confirmNewPassword: "confirmNewPassword",
	},
};

// re-export everything
export * from "@testing-library/react";
// override render method
export {
	render,
	tokenState,
	userTypeUserState,
	userTypeTradesmenState,
	idState,
	userDetailsState,
	handleSubmitState,
	handleChangeState,
	disabledState,
	disabledEmailState,
	buttonTextState,
	formDataState,
};
