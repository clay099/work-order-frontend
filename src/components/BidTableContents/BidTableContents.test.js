import React from "react";
import BidTableContents from "./BidTableContents";
import { render, bidData } from "../../testHelper/testHelper";
console.error = jest.fn();

const params = {
	projectId: 1,
	bidData: {
		1: { first_name: "first_name", last_name: "last_name", bid: 500 },
	},
};

const initialState = {
	login: { token: "token" },
};

describe("BidTableContents Component", () => {
	// smoke test
	test("renders without crashing", () => {
		render(<BidTableContents {...params} />, {
			initialState,
		});
	});

	// snapshot test
	test("matches snapshot", () => {
		const { asFragment } = render(<BidTableContents {...params} />, {
			initialState,
		});
		expect(asFragment).toMatchSnapshot();
	});

	test("value to be in the document", () => {
		const { getByText } = render(<BidTableContents {...params} />, {
			initialState,
		});
		const value = getByText("first_name");
		expect(value).toBeInTheDocument();
	});
});
