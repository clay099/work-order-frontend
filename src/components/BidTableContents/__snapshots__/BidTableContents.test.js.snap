// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidTableContents Component matches snapshot 1`] = `[Function]`;
